uberfire-project/uberfire-project-backend/src/main/java/org/guvnor/common/services/project/backend/server/WorkspaceProjectServiceImpl.java

```diff
@@ -280,6 +280,30 @@ public WorkspaceProject resolveProject(final Space space,
                               name);
     }
 
+    @Override
+    public WorkspaceProject resolveProject(final Space space,
+                                           final String projectName,
+                                           final String branchName) {
+
+
+        WorkspaceProject workspaceProject = resolveProject(space, projectName);
+
+        if (workspaceProject == null || branchName == null) {
+            return workspaceProject;
```

Comment:
I think it is best to throw a exception here to avoid a potential NPE in the caller.

Reply:
done